### DISCUSSION TITLE
Disabling Authorization during testing

### DISCUSSION THREAD

**Discussion by napondo** [loopbackio/loopback-next#8393] [Q&A]
I have implemented authorizations for controller endpoints using a [JWT strategy](https://loopback.io/doc/en/lb4/Authentication-tutorial.html). I would like to disable authentication/authorization when running particular tests. For example, I have a validation for users' email address uniqueness. I do not want to invoke authorization testing this code. How do I selectively disable authentication during testing?

**Comment by mrmodise** [comment]
I have in the past disabled authentication during test by not mounting the authentication component during testing. It might not address your specific use case but i think it gives you an idea unless there is a special decorator for this:

```ts
 if (process.env.NODE_ENV !== 'test') {
      this.component(AuthenticationComponent);
      registerAuthenticationStrategy(this, JWTAuthenticationStrategy);
    }
```

**Reply by napondo** [reply]
mrmodise Thanks for the prompt response.It is much appreciated.The example you gave would disable authorizations for all tests which would mean I cannot test if my authorization strategy is working.  Is it possible to implement your solution on a a test-by-test basis?

**Reply by mrmodise** [reply]
Thanks hence I cautioned it might not address your problem but it gives you an idea of what to disable. Let me do some digging to find out if there is another way.

@ritch @raymondfeng @marioestradarosa any thoughts?
